src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -983,6 +984,19 @@ HRESULT CorProfiler::ProcessReplacementCalls(
         continue;
       }
 
+      //   3) The target instruction is a constrained virtual method call (a constrained IL instruction followed by a callvirt IL instruction)
+      if (pInstr->m_opcode == CEE_CALLVIRT && pInstr->m_pPrev->m_opcode == CEE_CONSTRAINED) {
```

Comment:
Just so I'm clear, this is the only functional change, right? Everything else if for testing?

Did we consider just dropping the `CEE_CONSTRAINED` opcode?

Reply:
Yes this is the only functional change.

I determined that just dropping the `CEE_CONSTRAINED` opcode would also be lead to invalid code because earlier in the method, a **pointer** to the instance was pushed onto the stack, not the actual reference. As a result, we would have to do a significant amount of work to track down which `ldloca` or `ldloca.s` instruction was responsible for this `virtcall` and replace it. Here's the stack behavior for `constrained` calls from the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.constrained?view=netcore-3.1#remarks):

```
The state of the MSIL stack at this point must be as follows:
1) A managed pointer, ptr, is pushed onto the stack. The type of ptr must be a managed pointer (&) to thisType. Note that this is different from the case of an unprefixed callvirt instruction, which expects a reference of thisType.
2. Method arguments arg1 through argN are pushed onto the stack, just as with an unprefixed callvirt instruction.
```